shared/src/main/java/com/couchbase/lite/Join.java

```diff
@@ -13,12 +13,10 @@
  */
 package com.couchbase.lite;
 
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
-public class Join extends Query implements WhereRouter, OrderByRouter {
+public abstract class Join extends Query {
```

Comment:
Join shouldn't extends Query. There is Joins which extends Query and implements WhereRouter and OrderByRouter.